File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        levo = 0
        desno = len(nums)-1


        while levo<=desno:
            medijana = (levo + desno) // 2
            if nums[medijana]<target:
                levo=medijana+1
            elif nums[medijana]>target:
                desno=medijana-1
            else:
                return medijana
        return -1

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int levo =0, desno=0, duzina = s.length();
        Set<Character> set = new HashSet<>();
        int trenutno = 0, max=0;
        while (desno<duzina) {
            char karakter = s.charAt(desno);
            while (set.contains(karakter)) {
                set.remove(s.charAt(levo));
                levo++;
                trenutno-=1;
            }

            trenutno+=1;
            if (trenutno>max)
                max=trenutno;

            set.add(karakter);
            desno++;

        }
        return max;
    }
}
